MediaBrowser.MediaEncoding/Attachments/AttachmentExtractor.cs

```diff
@@ -230,13 +231,166 @@ public async Task<(MediaAttachment Attachment, Stream Stream)> GetAttachment(Bas
             MediaAttachment mediaAttachment,
             CancellationToken cancellationToken)
         {
+            await CacheAllAttachments(mediaPath, inputFile, mediaSource, cancellationToken).ConfigureAwait(false);
+
             var outputPath = GetAttachmentCachePath(mediaPath, mediaSource, mediaAttachment.Index);
             await ExtractAttachment(inputFile, mediaSource, mediaAttachment.Index, outputPath, cancellationToken)
                 .ConfigureAwait(false);
 
             return outputPath;
         }
 
+        private async Task CacheAllAttachments(
+            string mediaPath,
+            string inputFile,
+            MediaSourceInfo mediaSource,
+            CancellationToken cancellationToken)
+        {
+            var locks = new List<AsyncKeyedLockReleaser<string>>();
+            var extractableAttachmentIds = new List<int>();
+
+            try
+            {
+                foreach (var attachment in mediaSource.MediaAttachments)
+                {
+                    var outputPath = GetAttachmentCachePath(mediaPath, mediaSource, attachment.Index);
+
+                    var @lock = _semaphoreLocks.GetOrAdd(outputPath);
```

Comment:
Please rename this variable into something more descriptive

Reply:
Thanks, fixed.